connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -1999,6 +2004,81 @@ public void testErrorReporterConfigurationExceptionPropagation() {
         PowerMock.verifyAll();
     }
 
+    @Test
+    public void testPartitionCountInCaseOfPartitionRevocation() {
+        mockConsumer = new MockConsumer<>(OffsetResetStrategy.EARLIEST);
+        // Setting up Worker Sink Task to check metrics
+        workerTask = new WorkerSinkTask(
+                taskId, sinkTask, statusListener, TargetState.PAUSED, workerConfig, ClusterConfigState.EMPTY, metrics,
+                keyConverter, valueConverter, errorHandlingMetrics, headerConverter,
+                transformationChain, mockConsumer, pluginLoader, time,
+                RetryWithToleranceOperatorTest.NOOP_OPERATOR, null, statusBackingStore, Collections::emptyList);
+        // Subscribing to Topic = "test" with "MockHandleRebalance"
+        mockConsumer.subscribe(asList(TOPIC), new MockHandleRebalancePass());
```

Comment:
We shouldn't be subscribing directly here. Instead, the `WorkerSinkTask` instance should be issuing this call as part of its [initializeAndStart method](https://github.com/apache/kafka/blob/0390d5b1a24f6f4d0f431928dd009aeb76b3895d/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java#L311). This will also allow us to test the real consumer rebalance listener used by the `WorkerSinkTask` class, instead of a mocked listener.

I think the sequence of calls should be something like this:

- `MockConsumer::updateBeginningOffsets` (necessary because the `WorkerSinkTask` instance's rebalance listener will try to get the current position of its assignment in `onPartitionsAssigned`)
- `WorkerSinkTask::initialize` (you can use the `TASK_CONFIG` constant in the test class as the argument)
- `WorkerSinkTask::initializeAndStart`
- `assertSinkMetricValue`, `MockConsumer::rebalance` (as many times as you want)

Reply:
Thanks @C0urante , Yes this would be nice and I have added this step and removed the Mock Handle Rebalances.